backend/ops_api/ops/resources/agreements.py

```diff
@@ -76,6 +82,34 @@ class GrantAgreementRequestBody:
     notes: Optional[str] = None
 
 
+@dataclass
+class GrantAgreementPatchBody(GrantAgreementRequestBody):
+    name: Optional[str] = None
+    number: Optional[str] = None
+
+
+REQUEST_SCHEMAS = {
+    AgreementType.CONTRACT: {
+        "PUT": ContractAgreementRequestBody,
+        "PATCH": ContractAgreementPatchBody
+    },
+    AgreementType.GRANT: {
+        "PUT": GrantAgreementRequestBody,
+        "PATCH": GrantAgreementPatchBody
+    }
+}
+
+
+def pick_schema_class(agreement_type: AgreementType, method: str):
```

Comment:
would be useful for type hinting to have the return type here

Reply:
done